### PR TITLE
[Draft] High level design for OpenSearch Release CICD Infrastructure

### LINKED ISSUE #224: Test infrastructure high level design document

### PR DESCRIPTION
**Pull request by setiah**
The document covers the high level design approach for CICD infrastructure for OpenSearch artifact releases. The goal is to evolve this document as the finer details of design are worked out and use this as a go-to parent document for CICD.  

I know there is a [discussion around if we should have design documents in the same repo or a central repo](https://github.com/opensearch-project/.github/issues/28), but since the design is in progress and finer details are being worked out, it makes sense to have it closer to the code to keep it in sync with the implementation.

Signed-off-by: Himanshu Setia <setiah@amazon.com>

### Issues Resolved
#224 
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: bundle-workflow/python/test_workflow/integ_test_suite.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import os
```
Comment (setiah) [inline bundle-workflow/python/test_workflow/integ_test_suite.py:1]: please ignore this change. Will remove this in the next update.

**Comment by dblock** [comment]
Let's talk about where we want to put this? Why wouldn't this be the README of bundle-workflow?

**Comment by setiah** [comment]
> Let's talk about where we want to put this? Why wouldn't this be the README of bundle-workflow?

Sure, this was just a placeholder for now. Fine with whatever works best from discoverability perspective. :)
I personally feel having it in a `docs/` or `design-docs/` dir in the same repo makes it explicit and easy to discover. Having it as README under individual subfolders could lead to discoverability issues. In the longer run, it will be decided based on https://github.com/opensearch-project/.github/issues/28

**Comment by dblock** [comment]
> > Let's talk about where we want to put this? Why wouldn't this be the README of bundle-workflow?
> 
> Sure, this was just a placeholder for now. Fine with whatever works best from discoverability perspective. :)
> I personally feel having it in a `docs/` or `design-docs/` dir in the same repo makes it explicit and easy to discover. Having it as README under individual subfolders could lead to discoverability issues. In the longer run, it will be decided based on [opensearch-project/.github#28](https://github.com/opensearch-project/.github/issues/28)

I know we love design docs at Amazon, but all I am saying is that README in this repo = this design doc. I think you can refactor the text here into the README and I'd be happy to merge it. 

Do note that the end-to-end process glues parts such as bundle build, assemble and test. Each of the sections is a whole project by itself and lives in subfolders. So don't duplicate explaining what each of the subfolders does beyond a tl;dr. The meat should go into, for example, https://github.com/opensearch-project/opensearch-build/blob/main/bundle-workflow/README.md. The tll;dr and the link into README.

**Comment by dblock** [comment]
Can we close this?

**Comment by setiah** [comment]
> Can we close this?

Sure. I'll open another one as README at the discussed location.

**Pull request closed without merging**